Engine/AlgorithmManager.cs

```diff
@@ -482,12 +482,20 @@ public void Run(AlgorithmNodePacket job, IAlgorithm algorithm, IDataFeed feed, I
                 {
                     foreach (var update in timeSlice.ConsolidatorUpdateData)
                     {
+                        var resolutionTimeSpan = update.Target.Resolution.ToTimeSpan();
                         var consolidators = update.Target.Consolidators;
                         foreach (var consolidator in consolidators)
                         {
                             foreach (var dataPoint in update.Data)
                             {
-                                consolidator.Update(dataPoint);
+                                // Filter out data with resolution higher than the data subscription resolution.
+                                // This is needed to avoid feeding in higher resolution data, typically fill-forward bars.
+                                // It also prevents volume-based indicators or consolidators summing up volume to generate
+                                // invalid values.
+                                if (algorithm.Time == dataPoint.EndTime.RoundUp(resolutionTimeSpan))
+                                {
+                                    consolidator.Update(dataPoint);
```

Comment:
Would this be better placed on the time slice create method? Also,  for volume consolidators,  fill forward would also copy the volume property. Perhaps we can change the clone method to not populate volume when filling forward.